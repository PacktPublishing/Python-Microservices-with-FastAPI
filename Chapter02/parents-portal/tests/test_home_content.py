import pytest
from syrupy.extensions.single_file import (
    SingleFileAmberSnapshotExtension,
)


@pytest.mark.parametrize("path", ["/home", "/"])
def test_home_content(test_client, path, snapshot):
    """Test that /home and / return the welcome page."""
    response = test_client.get(path)
    assert response.status_code == 200
    assert response.text == snapshot


@pytest.mark.parametrize(
    "path", ["/home/en", "/home/fr", "/home/it", "/home/pt"]
)
def test_home_with_different_languages(
    test_client, path, snapshot
):
    response = test_client.get(path)
    assert response.status_code == 200
    assert response.text == snapshot(
        extension_class=SingleFileAmberSnapshotExtension
    )


def test_home_sets_and_uses_tracking_cookie(test_client):
    """Test that /home sets TRACKING cookie and shows welcome back message on repeat visit."""
    # First visit: should set cookie and show welcome
    response = test_client.get("/")
    assert response.status_code == 200
    assert "Welcome to the Portal" in response.text
    assert response.cookies.get("TRACKING") is not None

    # Second visit: should recognize cookie and show welcome back
    response = test_client.get("/")
    assert response.status_code == 200
    assert "Welcome back" in response.text
