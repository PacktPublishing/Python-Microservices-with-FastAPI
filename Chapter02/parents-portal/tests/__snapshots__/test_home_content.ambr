# serializer version: 1
# name: test_home_content[/]
  '''
  <html>
      <head>
          <title>Home</title>
      </head>
      <body>
          <h1>
              
              
                  Welcome to the Portal
              
          </h1>
      </body>
  </html>
  '''
# ---
# name: test_home_content[/home]
  '''
  <html>
      <head>
          <title>Home</title>
      </head>
      <body>
          <h1>
              
              
                  Welcome to the Portal
              
          </h1>
      </body>
  </html>
  '''
# ---
